pkg/daemon/daemon.go

```diff
@@ -237,11 +237,6 @@ func (dn *Daemon) isDesiredMachineState() (bool, error) {
 	if err != nil {
 		return false, err
 	}
-	// if the current annotation is equal to the desired annotation,
-	// system state is valid.
-	if strings.Compare(dcAnnotation, ccAnnotation) == 0 {
-		return true, nil
-	}
```

Comment:
This makes sense to me. Though... now I wonder about the opposite case. If the annotations *do* match, but `isDesiredMachineState` is `false`, then something has gone wrong, right? I guess this gets into #99, but it's slightly different: there, we're talking about repeatedly trying and failing to meet the desired config so that we can set the current config label to the same. Here, we've already in the past validated the node for that config (since the current config label already matches), but somehow the state doesn't match. I feel like we should just set it to degraded in that case?

So basically, it'd mean still doing this comparison but e.g. in `process` after the call to `isDesiredMachineState()`. Anyway, we can do that in a follow-up!

Reply:
my thought was that if there was a one-time change of a file under our control, that we can (and I think should) easily just change it back to what we want. if it happens multiple times right now, we don't do anything, but that was already true (maybe there are more cases of it now). I think doing some kind of backoff like described in #99 is the way to go in the long run, which I think would cover this case as well.